src/areastore.cpp

```diff
@@ -0,0 +1,355 @@
+/*
+Minetest
+Copyright (C) 2015 est31 <mtest31@outlook.com>
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation; either version 2.1 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/ 
+ 
+#include "areastore.h"
+#include "util/serialize.h"
+#include "util/container.h"
+
+#if USE_SPATIAL
+	#include <SpatialIndex.h>
+	#include <RTree.h>
+	#include <Point.h>
+#endif
+
+#define AST_SMALLER_EQ_AS(p, q) ((p.X <= q.X) && (p.Y <= q.Y) && (p.Z <= q.Z))
+
+#define AST_OVERLAPS_IN_DIMENSION(amine, amaxe, b, d) \
+	(!((amine.d > b->maxedge.d) || (amaxe.d < b->minedge.d)))
+
+#define AST_CONTAINS_PT(a, p) (AST_SMALLER_EQ_AS(a->minedge, p) && \
+	AST_SMALLER_EQ_AS(p, a->maxedge))
+
+#define AST_CONTAINS_AREA(amine, amaxe, b)         \
+	(AST_SMALLER_EQ_AS(amine, b->minedge) \
+	&& AST_SMALLER_EQ_AS(b->maxedge, amaxe))
+
+#define AST_AREAS_OVERLAP(amine, amaxe, b)                \
+	(AST_OVERLAPS_IN_DIMENSION(amine, amaxe, b, X) && \
+	AST_OVERLAPS_IN_DIMENSION(amine, amaxe, b, Y) &&  \
+	AST_OVERLAPS_IN_DIMENSION(amine, amaxe, b, Z))
+
+std::vector<std::string> get_areastore_typenames()
+{
+	std::vector<std::string> res;
+	res.push_back("Vector");
+#if USE_SPATIAL
+	res.push_back("LibSpatial");
+#endif
+	return res;
```

Comment:
Seeing as this is constant, you can just return a `*std::string[]` or so (assuming this is even needed).